### PR TITLE
Add vim and emacs keybindings support to CMS

### PR DESCRIPTION
**Pull request by ariponce**
Everytime i use the CMS i struggle due to my vim habits and seeing that Ace editor offers support for vim keybindings, i decided to add them to the CMS editor.

I've also added the Emacs because, why not?

They can be activated on the Editor preferences page.

### CODE REVIEW

File: modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js [hunk 1]
```diff
@@ -64,7 +64,7 @@
         showPrintMargin: true,
         highlightSelectedWord: false,
         hScrollBarAlwaysVisible: false,
-        readOnly: false
+        readOnly: false,
```
Comment (myst6re) [inline modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js]: This extra comma will cause an error on some old JS interpreter.

File: modules/backend/formwidgets/CodeEditor.php [hunk 2]
```diff
@@ -99,7 +104,8 @@ public function init()
             'fontSize',
             'margin',
             'theme',
-            'readOnly'
+			'readOnly',
+			'keyboard_binding'
```
Comment (myst6re) [inline modules/backend/formwidgets/CodeEditor.php]: Wrong indentation, you must be consistent.

**Comment by daftspunk** [comment]
Please use spaces instead of tabs. `keyboard_binding` should be `keyboardBinding` too I think.

**Comment by dshoreman** [comment]
+1. Would love to see support for vim bindings in the editor.

**Comment by dshoreman** [comment]
@ariponce A lot of the files still have a few tabs lingering in them. It's probably easier if you tell your editor to convert tabs to spaces rather than manually editing them.

@daftspunk Looking at the `keyboardBinding` var, are you sure? All other preferences use `snake_case` format, now it's the odd one out

**Comment by ariponce** [comment]
Ok, i think i converted all tabs to spaces.

Regarding the `keyboardBinding` var, i've used snake case on the lang file because i thought that was the standard (all other variables are snake case). I can change it to camel case if you want me to.

**Comment by ariponce** [comment]
Any news on this?

**Comment by daftspunk** [comment]
The editor page has since been rebuilt, sorry for the delay. Please resubmit at your earliest convenience.

**Pull request closed without merging**